src/Collector/RequestCollector.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Yiisoft\Yii\Debug\Collector;
+
+use Yiisoft\Yii\Web\Event\AfterRequest;
+use Yiisoft\Yii\Web\Event\BeforeRequest;
+
+final class RequestCollector implements CollectorInterface
+{
+    use CollectorTrait;
+
+    private string $requestUrl = '';
+    private string $requestMethod = '';
+    private bool $requestIsAjax = false;
+    private ?string $userIp = null;
+    private int $responseStatusCode = 200;
+
+    public function getCollected(): array
+    {
+        return [
+            'request_url' => $this->requestUrl,
+            'request_method' => $this->requestMethod,
+            'request_is_ajax' => $this->requestIsAjax,
+            'user_ip' => $this->userIp,
+            'response_status_code' => $this->responseStatusCode
+        ];
+    }
+
+    public function collect(object $event): void
+    {
+        if ($event instanceof BeforeRequest) {
+            $this->requestUrl = (string)$event->getRequest()->getUri();
+            $this->requestMethod = $event->getRequest()->getMethod();
+            $this->requestIsAjax = $event->getRequest()->getHeaderLine('X-Requested-With') === 'XMLHttpRequest';
+            $this->userIp = $event->getRequest()->getServerParams()['REMOTE_ADDR'] ?? null;
+        }
+        if ($event instanceof AfterRequest) {
+            $this->responseStatusCode = $event->getResponse()->getStatusCode();
+        }
+    }
+
+    private function reset(): void
```

Comment:
Is it used anywhere?

Reply:
Yes https://github.com/yiisoft/yii-debug/blob/d2c6faa98dbd62c3baf83f2ade8f0eb9137379bd/src/Collector/CollectorTrait.php#L16